v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/utils/ShardFileReader.java

```diff
@@ -189,13 +196,34 @@ public List<Shard> readForwardMigrationShardingConfig(String sourceShardsFilePat
     for (Map dataShard : dataShards) {
       List<Map> databases = (List) (dataShard.getOrDefault("databases", new ArrayList<>()));
 
+      String host = (String) (dataShard.get("host"));
+      if (databases.isEmpty()) {
+        LOG.warn("no databases found for host: {}", host);
+        throw new RuntimeException("no databases found for host: " + String.valueOf(host));
+      }
+
+      String password =
+          resolvePassword(
+              sourceShardsFilePath,
+              (String) dataShard.get("secretManagerUri"),
+              host,
+              (String) dataShard.get("password"));
+      if (password == null || password.isEmpty()) {
+        LOG.warn("could not fetch password for host: {}", host);
+        throw new RuntimeException(
+            "Neither password nor secretManagerUri was found in the shard file "
+                + sourceShardsFilePath
+                + "  for host "
+                + host);
+      }
+
       Shard shard =
           new Shard(
```

Comment:
We should verify that we dont print this struct anywhere.

Reply:
Yeah. Those place were removed as a part of the PR